lib/gollum-lib/committer.rb

```diff
@@ -75,14 +75,17 @@ def parents
     # path   - The String path to be added
     # data   - The String wiki data to store in the tree map.
     #
-    # Raises Gollum::DuplicatePageError if a matching filename already exists.
+    # Raises Gollum::DuplicatePageError if a matching filename already exists, unless force_overwrite is explicitly enabled.
     # This way, pages are not inadvertently overwritten.
     #
     # Returns nothing (modifies the Index in place).
-    def add_to_index(path, data, options = {})
+    def add_to_index(path, data, options = {}, force_overwrite = false)
       if index.current_tree
-        tree = @wiki.page_file_dir ? index.current_tree / @wiki.page_file_dir : index.current_tree
-        raise DuplicatePageError.new(path) if !page_path_scheduled_for_deletion?(index.tree, path) && tree.blobs.find {|blob| blob.name == path}
+        tree = index.current_tree
```

Comment:
Wait, should this not respect `page_file_dir`? (Compare the deleted line 84)
```ruby
tree = index.current_tree
tree = tree / @wiki.page_file_dir if @wiki.page_file_dir
```

Reply:
Both `index.current_tree` and `path` contain `page_file_dir` if it is set, so `tree = tree / path` works either way.